### PR TITLE
Update FSDP tutorial

### PR DESCRIPTION
**Pull request by davidberard98**
* rename default_auto_wrap_policy -> size_based_auto_wrap_policy (the interface was changed)
* indentation/imports/etc

### CODE REVIEW

**Comment by davidberard98** [comment]
re: size_based_auto_wrap_policy - note that older versions (before https://github.com/pytorch/pytorch/pull/76858) would still need default_auto_wrap_policy; do we need to mention this?

**Review by zhaojuanmao — APPROVED** [review]
Thanks for fixing it

**Comment by svekars** [comment]
> re: size_based_auto_wrap_policy - note that older versions (before [pytorch/pytorch#76858](https://github.com/pytorch/pytorch/pull/76858)) would still need default_auto_wrap_policy; do we need to mention this?

Since we don't version the tutorials build, I think it's a good idea to mention that and show how.

**Review by svekars — APPROVED** [review]

File: intermediate_source/FSDP_tutorial.rst [hunk 1]
```diff
@@ -62,11 +62,15 @@ We add the following code snippets to a python script “FSDP_mnist.py”.
 
 1.2  Import necessary packages
 
+.. note::
+    This tutorial is intended for pytorch versions 1.12 and later. If you are using 1.11 or earlier, replace all instances of `size_based_auto_wrap_policy` with `default_auto_wrap_policy`.
```
Comment (svekars) [inline intermediate_source/FSDP_tutorial.rst:66]:
```suggestion
    This tutorial is intended for PyTorch versions 1.12 and later. If you are using an earlier version, replace all instances of `size_based_auto_wrap_policy` with `default_auto_wrap_policy`.
```

**Pull request merged**